lib/srv/ctx.go

```diff
@@ -907,3 +912,14 @@ func newUaccMetadata(c *ServerContext) (*UaccMetadata, error) {
 		WtmpPath:   wtmpPath,
 	}, nil
 }
+
+// ComputeLockTargets computes lock targets inferred from a Server
+// and an IdentityContext.
+func ComputeLockTargets(s Server, id IdentityContext) ([]types.LockTarget, error) {
```

Comment:
nit: returned error is always `nil`
```suggestion
func ComputeLockTargets(s Server, id IdentityContext) []types.LockTarget {
```